### PR TITLE
Use Typography.js for global styles

### PR DESCRIPTION
**Pull request by KyleAMathews**
Thought you might be interested in seeing this.

Also adding a missing dependency + missing google fonts css.

### CODE REVIEW

File: source/website/Slides/0.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { TitleSlide } from '../../modules'
 
 const slide = () => (
   <TitleSlide>
-    <h1>React Presents: a slideshow framework</h1>
+    <h1 style={{color: 'white'}}>React Presents: a slideshow framework</h1>
```
Comment (bvaughn) [inline source/website/Slides/0.js]: Not a fan of this particularly. Should be set by the parent slide.

Why wouldn't the default for CSS be `color: inherit` ?

**Comment by bvaughn** [comment]
This looks cool. I dig it. 😄

Thanks for catching the missing `react-virtualized-select` dep too! Shame on me.

**Comment by bvaughn** [comment]
This PR also seems to cause funkiness with the CodeMirror highlighting:

screen shot 2016-10-24 at 8 35 32 am

**Comment by bvaughn** [comment]
This addresses some nits and linting issues, but the CodeMirror formatting is still kind of funky:

``` diff
diff --git a/source/modules/setDefaultTheme.js b/source/modules/setDefaultTheme.js
index 33e8e30..272f608 100644
--- a/source/modules/setDefaultTheme.js
+++ b/source/modules/setDefaultTheme.js
@@ -1,19 +1,27 @@
 import { injectGlobal } from 'styled-components'
 import Typography from 'typography'

+const headerOverrideStyle = {
+  color: 'inherit'
+}
+
 const theme = {
   baseFontSize: '20px',
   bodyFontFamily: ['Droid Sans', 'sans-serif'],
   headerFontFamily: ['Yanone Kaffeesatz', 'sans-serif'],
   headerWeight: 400,
   scaleRatio: 2.5,
-  overrideStyles: ({rhythm}) => ({
+  overrideStyles: ({ rhythm }) => ({
+    h1: headerOverrideStyle,
+    h2: headerOverrideStyle,
+    h3: headerOverrideStyle,
+    h4: headerOverrideStyle,
     li: {
-      marginBottom: 0,
+      marginBottom: 0
     },
     '.ReactCodeMirror': {
-      marginBottom: rhythm(1),
-    },
+      marginBottom: rhythm(1)
+    }
   })
 }

@@ -22,6 +30,7 @@ const typography = new Typography(theme)
 export default () => {
   injectGlobal`
     ${typography.toString()}
+
     html, body, #root {
       height: 100%;
     }
```

**Comment by KyleAMathews** [comment]
Fixed the codemirror styles — the codemirror.css overrode the styles from typography.js so I overrode it back. Yeah, css battles!

re: `color: inherit`. This isn't a good default as quite often you want the header color to be different from the body color. Headers are often darker than the body text for example, Typography.js lets you set them both in the API.

**Comment by KyleAMathews** [comment]
It looks like codemirror is doing something funky with their scrollbars as well. I looked at that a bit but didn't find the root cause.

**Comment by bvaughn** [comment]
> re: color: inherit. This isn't a good default as quite often you want the header color to be different from the body color. Headers are often darker than the body text for example, Typography.js lets you set them both in the API.

So long as it's overridable (which it is) then users can always set custom header colors. But the default should be `inherit` IMO.

Put it another way: If I want a custom header color, I'm already going to have to specify it, so making `inherit` the default adds no additional work for people wanting custom colors. However, if I want the same color (which is at least a fairly common use-case) I wouldn't have to do anything _without_ Typography.js. _With_ it though I have to specify an override. 😁 

**Edit**: I think this same argument applies to thinks like `box-sizing`. Even though (in my opinion) `border-box` is _always_ what I want, it's still more flexible to set that style on the root and let `*` `inherit` it.

File: source/modules/setDefaultTheme.js [hunk 2]
```diff
@@ -1,65 +1,49 @@
...
+  baseFontSize: '20px',
+  bodyFontFamily: ['Droid Sans', 'sans-serif'],
+  headerFontFamily: ['Yanone Kaffeesatz', 'sans-serif'],
+  headerWeight: 400,
+  scaleRatio: 2.5,
+  plugins: [
+    new CodePlugin()
+  ],
+  overrideStyles: ({scale, rhythm}) => ({
+    'h1,h2,h3,h4,h5,h6': {
```
Comment (bvaughn) [inline source/modules/setDefaultTheme.js:15]: Oh nice. 👍 This is better than what I had.

**Review by bvaughn — APPROVED** [review]

**Comment by bvaughn** [comment]
Thanks for the PR @KyleAMathews. I'm down to use it. Seems like a pretty slick utility.

**Comment by KyleAMathews** [comment]
Mmmm yeah I see what you mean now. Yeah, setting the header color to be darker than the body color is a very reasonable design choice but isn't necessary and can be a confusing default. Yeah I think I agree now. Set a body text color by default and the header color just inherits from that. Still make it easy to set the header color to something with the API but make it opt-in.

**Comment by KyleAMathews** [comment]
Great!

**Comment by bvaughn** [comment]
👍

**Comment by KyleAMathews** [comment]
re your comment on `box-sizing` — isn't this what Typography.js does? Perhaps not understanding you correctly...

**Comment by bvaughn** [comment]
Sorry, I was just giving a related example of using inherit for a style _even when it's the "right" style_

**Comment by KyleAMathews** [comment]
ah gotcha :-)

**Pull request merged**